discovery.rst

```diff
@@ -1,12 +1,16 @@
 Discovery
 =========
 
-The discovery service is a set of static classes which allows to find and use installed resources.
+The discovery service allows to find and use installed resources.
+Under the hood it uses `Puli`_ for the discovery logic. All of our packages provide Puli resources which means they
+could even be used without this convenience wrapper. (For example HttplugBundle only relies on Puli).
+However, in some cases you need to be able to statically access the discovery layer (when DI is not an option).
 This is useful in libraries that want to offer zero-configuration services and rely only on the virtual packages.
```

Comment:
All of our packages provide Puli resources. Discovery is simply a convenience wrapper to statically access clients and factories for when Dependency Injection is not an option. Discovery is useful in libraries that want to offer zero-configuration services relying on the virtual packages [link to virtual packages doc]. If you have Dependency Injection available, using Puli directly is more elegant (see for example the Symfony ``HttplugBundle``).